test/e2e/operator_test.go

```diff
@@ -258,25 +305,25 @@ func TestContourNodePortService(t *testing.T) {
 	t.Logf("deleted contour %s/%s", operatorNs, testName)
 
 	// Ensure the envoy service is cleaned up automatically.
-	if err := waitForServiceDeletion(ctx, kclient, 3*time.Minute, specNs, "envoy"); err != nil {
-		t.Fatalf("failed to delete envoy service %s/envoy: %v", specNs, err)
+	if err := waitForServiceDeletion(ctx, kclient, 3*time.Minute, cfg.SpecNs, "envoy"); err != nil {
+		t.Fatalf("failed to delete envoy service %s/envoy: %v", cfg.SpecNs, err)
 	}
-	t.Logf("cleaned up envoy service %s/envoy", specNs)
+	t.Logf("cleaned up envoy service %s/envoy", cfg.SpecNs)
 
 	// Delete the operand namespace since contour.spec.namespace.removeOnDeletion
 	// defaults to false.
-	if err := deleteNamespace(ctx, kclient, 5*time.Minute, specNs); err != nil {
-		t.Fatalf("failed to delete namespace %s: %v", specNs, err)
+	if err := deleteNamespace(ctx, kclient, 1*time.Minute, cfg.SpecNs); err != nil {
+		t.Fatalf("failed to delete namespace %s: %v", cfg.SpecNs, err)
 	}
-	t.Logf("observed the deletion of namespace %s", specNs)
+	t.Logf("observed the deletion of namespace %s", cfg.SpecNs)
 }
 
 func TestContourClusterIPService(t *testing.T) {
 	testName := "test-clusterip-contour"
 	cfg := objcontour.Config{
 		Name:        testName,
 		Namespace:   operatorNs,
-		SpecNs:      specNs,
+		SpecNs:      fmt.Sprintf("%s-clusterip", specNs),
```

Comment:
Seems like it would be worth it to have a helper to generate a new random namespace name and to DRY this out with some more structure, here or in the future